projects/noloan2/app/src/main/java/hasadna/noloan2/FirestoreClient.java

```diff
@@ -0,0 +1,88 @@
+package hasadna.noloan2;
+
+import android.support.annotation.NonNull;
+import android.util.Base64;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import hasadna.noloan2.protobuf.SMSProto.SmsMessage;
+
+
+public class FirestoreClient {
+  FirebaseFirestore client;
+  
+  String collection;
+  
+  public FirestoreClient(String collection) {
```

Comment:
I think you should hardcode `collection`'s value in this class, since this only works for the SmsMessage collection right now.
I would call it `SMS_MESSAGE_COLLECTION`.